libbitdht/src/bitdht/bdboot_generate.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+<<LICENSE
+
+Copyright (C) 2020  Gioacchino Mazzurco <gio@eigenlab.org>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+LICENSE
+
+<<README
+Generate a clean DHT bootstrap node list.
+Feeds on previously known nodes from standard input and a few well known
+mainline DHT bootstrap nodes. Prints only nodes that appears to be active to a
+quick nmap check. Make sure your internet connection is working well before
+using this.
+
+Example usage:
+--------------------------------
+cat bdboot.txt | bdboot_generate.sh | tee /tmp/bdboot_generated.txt
+cat /tmp/bdboot_generated.txt | sort -u > bdboot.txt
+--------------------------------
+README
+
+function check_dht_host()
+{
+	mHost="$1"
+	mPort="$2"
+	
+	sudo nmap -oG - -sU -p $mPort $mHost | grep open | \
```

Comment:
Seems overkill to go admin for testing each server. Maybe you can keep sudo outside the script, so you call sudo once for the entire script. 

Also nmap is known to be pretty aggressive.

Reply:
`nmap` needs to be run as root to run that specific test, with the options provided it just test that UDP port for some kind of answer, so the test which is run is not aggressive ;)
I agree it is not the most elegant solution but it is the best we have ATM ;)
Putting `sudo` there is to minimize the code that run as root, otherwise the whole script would run as root :-|